### ISSUE TITLE
Messenger doesn't handle child-process getting killed unexpectedly

### ISSUE THREAD

**Issue by Rakdar** [WyriHaximus/reactphp-child-process-messenger#14]
Hi WyriHaximus,

i found an edge-case which this library doesnt yet support.

If you have a child process running, which for example exceeds the php memory_limit-value, it doesn't get caught by the messenger.

Here is some example code to reproduce the problem:

### ChildProcess class
```php
<?php

namespace Application\ChildProcess;

use React\EventLoop\LoopInterface;
use WyriHaximus\React\ChildProcess\Messenger\ChildInterface;
use WyriHaximus\React\ChildProcess\Messenger\Messenger;

class Overflow implements ChildInterface
{
    public static function create(Messenger $messenger, LoopInterface $loop)
    {
        $messenger->registerRpc('overflow', function () {
            ini_set('memory_limit','20M');

            $string = '';
            while (true) {
                $string .= '0123456789';
            }
        });
    }
}
```

### main process that never realizes the child process has been killed
```php
<?php

use Application\ChildProcess\Overflow;
use React\EventLoop\Factory;
use WyriHaximus\React\ChildProcess\Messenger\Factory as MessengerFactory;
use WyriHaximus\React\ChildProcess\Messenger\Messenger;
use WyriHaximus\React\ChildProcess\Messenger\Messages\Factory as MessageFactory;

require '../vendor/autoload.php';

$loop = Factory::create();

MessengerFactory::parentFromClass(Overflow::class, $loop)->then(function (Messenger $messenger) {
    $messenger->rpc(MessageFactory::rpc('overflow', []))->then(
        function () {
            echo "Obviously never gets here";
        },
        function () {
            echo "But also never gets here";
        }
    );
});

$loop->run();
```

**Comment by WyriHaximus** [comment]
Good catch, I'll look into a fix for this 👍

**Comment by WyriHaximus** [comment]
Been thinking about this, and crashing in the middle of an RPC call will reject the RPC call with an error.

**Comment by Rakdar** [comment]
Any idea what should happen if there is currently no RPC running?

If for example a simple php time_limit is exceeded and the process stops. All future RPC have to be rejected immediately, but maybe also the ```parentFormClass()->then()``` promise should be rejected?

Then it is in the hands of the user to catch and deal with corresponding Exception.

**Comment by WyriHaximus** [comment]
So I have a PoC locally that addressed this and what it does is emit an error and also reject all outstanding RPC calls. Haven't looked at the factory part yet

**Comment by kelunik** [comment]
@WyriHaximus I guess it should respawn automatically, otherwise things will probably be fragile, because people don't check and retry with a new instance.

**Comment by WyriHaximus** [comment]
@kelunik it shouldn't because we don't know we don't know what people run in it and how it affects their application. Things can go wrong on an epic scale for that application if we just blindly respawn and try again for a few times. Failing by rejecting all outstanding RPC's and emitting an error on the `Messenger` will give them plenty opportunity to deal with those errors and recover in the for their application correct way.

**Comment by WyriHaximus** [comment]
FYI this has been fixed in [`2.8.1`](https://github.com/WyriHaximus/reactphp-child-process-messenger/releases/tag/2.8.1)

**Comment by Rakdar** [comment]
Great Job! Thank you :+1:
